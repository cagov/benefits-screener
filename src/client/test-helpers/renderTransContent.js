import React from "react";
import { mount } from "enzyme";
import { I18nextProvider } from "react-i18next";
import i18n from "./i18nForTests";

function renderTransContent(Component, props = {}) {
  const wrapper = mount(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  // Filter out the very verbose I18nextProvider component wrapper
  return wrapper.find(Component);
}

module.exports = renderTransContent;
