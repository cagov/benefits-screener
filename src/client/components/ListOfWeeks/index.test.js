import renderTransContent from "../../test-helpers/renderTransContent";
import Component from "./index";

describe("<ListOfWeeks />", () => {
  it("renders the component", async () => {
    const wrapper = renderTransContent(Component, {
      weeksToCertify: [0, 5],
    });

    expect(wrapper).toMatchSnapshot();
  });

  it("with checks", async () => {
    const wrapper = renderTransContent(Component, {
      weeksToCertify: [1],
      showChecks: true,
    });

    expect(wrapper).toMatchSnapshot();
  });
});
