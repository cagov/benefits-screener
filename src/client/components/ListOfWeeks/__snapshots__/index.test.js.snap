// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListOfWeeks /> renders the component 1`] = `
<ListOfWeeks
  weeksToCertify={
    Array [
      0,
      5,
    ]
  }
>
  <Alert
    className="d-flex"
    closeLabel="Close alert"
    key="weekToCertify0"
    show={true}
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "displayName": "Fade",
        "render": [Function],
      }
    }
    variant="secondary"
  >
    <Fade
      appear={false}
      in={true}
      mountOnEnter={false}
      timeout={300}
      unmountOnExit={true}
    >
      <Transition
        addEndListener={[Function]}
        appear={false}
        enter={true}
        exit={true}
        in={true}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={300}
        unmountOnExit={true}
      >
        <div
          className="fade d-flex alert alert-secondary show"
          role="alert"
        >
          <div
            className="flex-fill"
          >
            <Trans
              i18nKey="retrocerts-week-list-item"
              t={[Function]}
              values={
                Object {
                  "endDate": "02/08/2020",
                  "startDate": "02/02/2020",
                  "weekForUser": 1,
                }
              }
            >
              <strong
                key="strong-0"
              >
                Week 1
              </strong>
              : Sunday, 02/02/2020 – Saturday, 02/08/2020
            </Trans>
          </div>
        </div>
      </Transition>
    </Fade>
  </Alert>
  <Alert
    className="d-flex"
    closeLabel="Close alert"
    key="weekToCertify5"
    show={true}
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "displayName": "Fade",
        "render": [Function],
      }
    }
    variant="secondary"
  >
    <Fade
      appear={false}
      in={true}
      mountOnEnter={false}
      timeout={300}
      unmountOnExit={true}
    >
      <Transition
        addEndListener={[Function]}
        appear={false}
        enter={true}
        exit={true}
        in={true}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={300}
        unmountOnExit={true}
      >
        <div
          className="fade d-flex alert alert-secondary show"
          role="alert"
        >
          <div
            className="flex-fill"
          >
            <Trans
              i18nKey="retrocerts-week-list-item"
              t={[Function]}
              values={
                Object {
                  "endDate": "03/14/2020",
                  "startDate": "03/08/2020",
                  "weekForUser": 2,
                }
              }
            >
              <strong
                key="strong-0"
              >
                Week 2
              </strong>
              : Sunday, 03/08/2020 – Saturday, 03/14/2020
            </Trans>
          </div>
        </div>
      </Transition>
    </Fade>
  </Alert>
</ListOfWeeks>
`;

exports[`<ListOfWeeks /> with checks 1`] = `
<ListOfWeeks
  showChecks={true}
  weeksToCertify={
    Array [
      1,
    ]
  }
>
  <Alert
    className="d-flex"
    closeLabel="Close alert"
    key="weekToCertify1"
    show={true}
    transition={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "appear": false,
          "in": false,
          "mountOnEnter": false,
          "timeout": 300,
          "unmountOnExit": false,
        },
        "displayName": "Fade",
        "render": [Function],
      }
    }
    variant="secondary"
  >
    <Fade
      appear={false}
      in={true}
      mountOnEnter={false}
      timeout={300}
      unmountOnExit={true}
    >
      <Transition
        addEndListener={[Function]}
        appear={false}
        enter={true}
        exit={true}
        in={true}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={300}
        unmountOnExit={true}
      >
        <div
          className="fade d-flex alert alert-secondary show"
          role="alert"
        >
          <div
            className="flex-fill"
          >
            <Trans
              i18nKey="retrocerts-week-list-item"
              t={[Function]}
              values={
                Object {
                  "endDate": "02/15/2020",
                  "startDate": "02/09/2020",
                  "weekForUser": 1,
                }
              }
            >
              <strong
                key="strong-0"
              >
                Week 1
              </strong>
              : Sunday, 02/09/2020 – Saturday, 02/15/2020
            </Trans>
          </div>
          <img
            alt="checkmark"
            className="checkmark"
            src="/images/check-circle-fill.svg"
          />
        </div>
      </Transition>
    </Fade>
  </Alert>
</ListOfWeeks>
`;
